sdk/core/Azure.Core/src/Pipeline/Internal/HttpPipelineTransportPolicy.cs

```diff
@@ -24,22 +25,20 @@ public override async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory
         {
             Debug.Assert(pipeline.IsEmpty);
 
-            await _transport.ProcessAsync(message).ConfigureAwait(false);
+            await _transport.ProcessAsync(message as PipelineMessage).ConfigureAwait(false);
```

Comment:
This calls through to the base PipelineTransport implementation that handles the response buffering now.